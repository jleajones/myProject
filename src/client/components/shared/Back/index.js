import React from 'react';
import Typography from '@material-ui/core/Typography';
import KeyboardBackspace from '@material-ui/icons/KeyboardBackspace';

import styles from './stylesheet.css';
import Link from '@components/Link';
import PATHS from '@constants/paths';

const Back = () => (
  <div className={styles.container}>
    <Typography variant="h6">
      <Link to={{ pathname: PATHS.home.path }} label={'back'} color={'primary'}>
        <KeyboardBackspace />
        <span className={styles.text}>Back</span>
      </Link>
    </Typography>
  </div>
);

export default Back;
