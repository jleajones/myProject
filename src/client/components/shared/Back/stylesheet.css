.container {
  padding-top: 24px;
}

.text {
  display: inline-block;
  vertical-align: text-bottom;
}
