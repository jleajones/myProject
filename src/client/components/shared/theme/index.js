import React from 'react';
import { createMuiTheme } from '@material-ui/core/styles';
import { ThemeProvider as MuiThemeProvider } from '@material-ui/styles';
import { ThemeProvider } from 'styled-components';
import PropTypes from 'prop-types';

export const theme = createMuiTheme({});

export const themeS = {
  colors: {
    light: 'white',
    lightBg: '#cccccc',
    primary: 'red'
  }
};

/**
 *
 * @param children
 * @param muiTheme
 * @param styledTheme
 * @returns {*}
 * @constructor
 */
const MyThemeProvider = ({ children, muiTheme, styledTheme }) => (
  <MuiThemeProvider theme={muiTheme}>
    <ThemeProvider theme={styledTheme}>{children}</ThemeProvider>
  </MuiThemeProvider>
);

MyThemeProvider.defaultProps = {
  muiTheme: theme,
  styledTheme: themeS
};

MyThemeProvider.propTypes = {
  children: PropTypes.node.isRequired,
  muiTheme: PropTypes.shape(),
  styledTheme: PropTypes.shape()
};

export default MyThemeProvider;
