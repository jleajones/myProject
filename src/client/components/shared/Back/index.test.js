import React from 'react';
import { render } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import 'jest-styled-components';
import PATHS from '@constants/paths';
import { themeS } from '@shared/theme';
import Back from '.';

test('Renders Back Button', () => {
  const component = render(
    <MemoryRouter initialEntries={[PATHS.driveWithUs.path]}>
      <Back theme={themeS} />
    </MemoryRouter>
  );

  expect(component).toMatchSnapshot();
});
