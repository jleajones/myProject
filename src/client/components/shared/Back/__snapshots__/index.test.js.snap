// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Back Button 1`] = `
.c0 {
  padding-top: 24px;
}

.c1 {
  color: white;
}

.c2 {
  display: inline-block;
  vertical-align: text-bottom;
}

<div
  className="c0"
>
  <h6
    className="MuiTypography-root MuiTypography-h6"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <span
        className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <span
          className="c1"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M21 11H6.83l3.58-3.59L9 6l-6 6 6 6 1.41-1.41L6.83 13H21z"
            />
          </svg>
          <span
            className="c2"
          >
            Back
          </span>
        </span>
      </span>
    </a>
  </h6>
</div>
`;
